### DISCUSSION TITLE
Possible bug in 5.23 discovered when I retrofitted gsd5

### DISCUSSION THREAD

**Discussion by Ca1vino** [TiddlyWiki/TiddlyWiki5#5534] [General]
The following might be of interest. It's about what appears to me to be a subtle bug in 5.23.

While retrofitting gsd5 to TW5.23, I found that tiddler folding was broken. Fixing that, I found that the feature that scrolls the story river to the last clicked tiddler title was also broken, and the story river would not auto-scroll.

After much tracing around, discovered it depends on something very subtle, which is the presence of any whitespace or enter characters around the markup at the beginning of the code in ViewTemplate.  It looks so natural to add whitespace between the close and open tags but it totally confuses code in PopStoryView.prototype.navigateTo.

DESCRIPTION
There's code in list.js module, function listwidget.findlistitem, that searches for the title in the DOM. It gets confused if the DOM of the storyriver tiddler list starts with a text element, in which case it doesn't dispatch the click event. The malformed DOM can be generated by any whitespace at the start of the ViewTemplate definition, e.g. simply by return characters between <$vars...>, <$tiddler...>, and the first <div>. This will result in failure to scroll properly to the clicked title's tiddler in the storyriver (code is in PopStoryView.prototype.navigateTo in module pop.js)

**Comment by Jermolene** [comment]
Hi @Ca1vino that is indeed the problem. Another way of looking at it is that the story views rely on each item generated by the list widget being wrapped in a single DOM element, with no intervening text elements, but that the list widget itself is capable of operating without any DOM wrapper around the list items.

One solution might be to separate the story river functionality from the generic list widget functionality, and have the replacement story river mechanism automatically add the wrapper elements.
